internal/conf/vault/vault.go

```diff
@@ -0,0 +1,179 @@
+//go:build edgex
+// +build edgex
```

Comment:
oh one more thing here. I gated this behind the edgex build flag. I need to try building without specifying the flag to ensure it still builds. Also looking for feedback on whether or not this is the preferred approach

Reply:
Actually, the whole openziti thing here depends on EdgeX's setting. I would rather guard them all behind the edgex build flag.

Reply:
i have put all edgex/openziti related features behind a build tag. please review the mechanism I went with. I think it makes sense... see what you think. marking resolved.